src/main/kotlin/racingame/domain/Cars.kt

```diff
@@ -0,0 +1,12 @@
+package racingame.domain
+
+class Cars(
+    values: List<Car> = emptyList()
+) {
+    var values = values
+        private set
```

Comment:
private set을 통해 외부의 접근을 막았지만, 
외부에서 데이터의 변경할수 있는 구조예요! 
방어적 복사를 고민해보셔도 좋을거같아요!

```suggestion
class Cars(
    private val values: List<Car> = emptyList()
) {
    val history get() = values.map{ it.copy ... }
```

Reply:
> RacingGame 클래스에서 View 의존성을 없애기 위해 경주 이력을 따로 저장해 관리하는 것으로 결정하였습니다!
> 하지만 이렇게 될 경우 Car의 과거 이력들을 저장하기 위해 깊은 복사가 필요했고 data class의 copy를 사용할까 했으나 이력 저장을 위해 data class로 바꿔야 할까 하는 의문이 들어 Cars 클래스에 Car를 추가할 때 추가되는 Car에 대해 새로 객체를 만들어주는 방식을 선택하였습니다.
> data class를 사용할 경우 copy와 보일러플레이트 코드를 줄여주는 장점 외에는 커스텀 getter, setter등을 사용하기 까다로운 문제가 있어 이러한 방식을 선택하였는데 혹시 더 좋은 방법이 있는지 궁금합니다!

좋은 고민을 해주셨네요!! 
우선 현재의 구조상 외부에서 Cars의 values에 접근하여 수정할수 있는 구조는 위험한 구조는 아닐까요?
방어적 복사가 필요한 상황은 아닐까 생각이 들어요! 

data class는 공식문서처럼 데이터를 보유하는 목적의 클래스이기 때문에,
(https://kotlinlang.org/docs/data-classes.html)
copy의 목적으로 data class를 사용하기보다는,
Car객체의 copy함수를 만들어서 사용하거나, 
Car객체를 데이터보유의 책임만 가지도록 리팩터링되도록 해도 좋을거같아요!
구현에는 정답이 없으니, 송이님이 원하는 구조를 고민해도 좋을 거같습니다!

다음 미션들을 진행하면서 배우시겠지만, 불변객체나 일급 컬렉션 등을 적용해보는것도 좋을거같긴해요!